src/lakefs_spec/spec.py

```diff
@@ -90,6 +91,17 @@ def parse(path: str) -> tuple[str, str, str]:
     repo, ref, resource = results.groups()
     return repo, ref, resource
 
+def check_branch_exists_or_create(client, repository, new_branch_name:str, source_branch_name:str = 'main'):
+        try:
+            #TODO: (m.mynter) How to get the source branch in implicit branch creation? 
```

Comment:
Yes, I think it makes sense to have it there. AFAIK there is no possibility to get the previously checked out branch for lakeFS. Plus, maybe I want another branch as source branch.

Reply:
Makes sense. So between

- memoizing the last checked out branch for each resource (presumably in a cache?),
- giving the option of a default file system-wide base branch,
- and always branching off of main,

option 2) is the best IMO. Making `main` the default is probably good enough for 90% of cases, but setting it as an attribute is good practice anyway.

What do you think? If you want to go ahead without, it can also go into a follow-up.

Reply:
- [Change](https://github.com/appliedAI-Initiative/lakefs-spec/pull/15/files#diff-f8235da430cb8c0f32975a91dcd263e02044b2621322a7e6c97182b53649ea1eR457) for `LakeFSFile` 
- - [Change](https://github.com/appliedAI-Initiative/lakefs-spec/pull/15/files#diff-f8235da430cb8c0f32975a91dcd263e02044b2621322a7e6c97182b53649ea1eR143) for `LakeFSFileSystem`